src/ansys/geometry/core/designer/geometry_commands.py

```diff
@@ -842,3 +846,68 @@ def update_fill_pattern(
         )
 
         return result.result.success
+
+    @protect_grpc
+    @min_backend_version(25, 2, 0)
+    def split_body(
+        self,
+        bodies: List["Body"],
+        plane: Plane,
+        slicers: Union["Edge", List["Edge"], "Face", List["Face"]],
+        faces: List["Face"],
+        extendfaces: bool,
+    ) -> bool:
+        """Split bodies with a plane, slicers, or faces.
+
+        Parameters
+        ----------
+        bodies : List[Body]
+            Bodies to split
+        plane : Plane
+            Plane to split with
+        slicers : Edge | list[Edge] | Face | list[Face]
+            Slicers to split with
+        faces : List[Face]
+            Faces to split with
+        extendFaces : bool
+            Extend faces if split with faces
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+        from ansys.geometry.core.designer.body import Body
+
+        check_type_all_elements_in_iterable(bodies, Body)
+
+        for body in bodies:
+            body._reset_tessellation_cache()
+
+        plane_item = None
+        if plane is not None:
+            plane_item = plane_to_grpc_plane(plane)
+
+        slicer_items = None
+        if slicers is not None:
+            slicer_items = [slicer._grpc_id for slicer in slicers]
+
+        face_items = None
+        if faces is not None:
+            face_items = [face._grpc_id for face in faces]
```

Comment:
Before iterating on the method inputs we should be performing some type checking since we are using loose type hints (i.e. "Face" instead of Face + checker decorator)